hat240.sh

```diff
@@ -8,72 +8,94 @@
 # $1 = Test File
 # $2 = Test Count
 # $3 = Target Score
-# $4 = Files For Saving on Failure
 #
 
 if [ -z "$1" ]; then
     read -e -p "Where is the test file? " homework_test_file
 else
     homework_test_file=$1
 fi
+
+if [ ! -f $homework_test_file ]; then
+    echo "File not found. Please ensure you have the correct path."
+    exit 1
+fi
+
 if [ -z "$2" ]; then
     read -e -p "How many times would you like to test? " test_count
 else
     test_count=$2
 fi
+
+if [ $test_count -le 0 ]; then
+    echo "Can't execute less than 1 time"
+    exit 1
+fi
+
 if [ -z "$3" ]; then
     target_score=100
 else
     target_score=$3
 fi
-if [ -z "$4" ]; then
-    files_for_saving="*Bogus* *Input* *Output* *Result*"
-else
-    files_for_saving=$4
-fi
 
 tests_failed=0
 minimum_score=100
 minimum_score_failure=0
 
-rm -rf test_failures
-
-mkdir -p test_failures
+if [ -d test_failures ]; then
+    rm -rf test_failures/*
+else
+    mkdir -p test_failures
+fi
 touch "test_failures/failures.txt"
 
+cd test_failures
+
+bar="=============================="
+bar_length=${#bar}
+echo
+
 for i in `seq 1 $test_count`; do
-    $($homework_test_file > test_output.txt)
+    # checks for directory 
+    if [ ! -d "failure$tests_failed" ]; then
+        mkdir "failure$tests_failed"
+    fi
+
+    # move into test failed and copy test file
+    cd "failure$tests_failed"
+    cp ../../$homework_test_file test_file
+
+    # execute and calculate score
+    $(./test_file > test_output.txt)
     score=$(cat test_output.txt | grep "total score" | grep -E -o "[0-9]+")
 
-    if [ $score = $target_score ] || [ $score -gt $target_score ]; then
-        rm -f $files_for_saving
+    if [ $score -eq $target_score ] || [ $score -gt $target_score ]; then
+        cd ..
+	rm -rf "Failure$tests_failed"
```

Comment:
I would highly recommend paying attention to your capitalization of directories. The directory you are deleting here is not the same directory that was created and you moved from.